docs: Make links out of @see tags

JeanMeche opened the pull request:
See individual commits

This commit is part of the work for #50097 to improve the linking on the online documentation.

## PR Type
What kind of change does this PR introduce?

- [x] angular.io application / infrastructure changes

JeanMeche commented on packages/router/src/provide_router.ts:
```diff
@@ -624,9 +624,9 @@ export type NavigationErrorHandlerFeature =
- * @see `inject`
- * @see `EnvironmentInjector#runInContext`
+ * @see {@link NavigationError}
+ * @see {@link core/inject}
```
fyi, this is required because `inject` is ambiguous.

JeanMeche committed:
docs(docs-infra): create code block for links to methods

Before this commit, links to methods had to rely on a custom title (or the method name would be displayed).

With this commit, {@link MyClass#myMethod} will ouput a link as `MyClass#myMethod`.
This has a 2nd advantage : We can now rely on the validation by the `checkAnchorLinksProcessor` to ensure the method actually exist.

This commit is part of #50110 to improve linking in the documentation.

JeanMeche committed:
docs(router): Make links out of @see tags

This commit is part of the work for #50097 to improve the linking on the online documentation.

JeanMeche committed:
docs(platform-browser): Make links out of @see tags

This commit is part of the work for #50097 to improve the linking on the online documentation.

JeanMeche committed:
docs(forms): Make links out of @see tags

This commit is part of the work for #50097 to improve the linking on the online documentation.

JeanMeche committed:
docs(core): Make links out of @see tags

This commit is part of the work for #50097 to improve the linking on the online documentation.

JeanMeche committed:
docs(common): Make links out of @see tags

This commit is part of the work for #50097 to improve the linking on the online documentation.

JeanMeche committed:
docs(animations): Make links out of @see tags

This commit is part of the work for #50097 to improve the linking on the online documentation.

AndrewKushnir commented:
> Is this PR acceptable with the multiple scopes or should it be split to ease the review ?

@JeanMeche I think a single PR makes sense in this case, since the changes are mostly identical in all packages. We'd probably need a global docs approver though to review this PR (cc @josephperrott @devversion).

devversion commented on packages/animations/src/animation_builder.ts:
```diff
@@ -59,7 +59,7 @@ export abstract class AnimationBuilder {
    * @param animation A reusable animation definition.
    * @returns A factory object that can create a player for the defined animation.
-   * @see `animate()`
+   * @see {@link animate}
```
Should we enforce this pattern somehow in Dgeni? i.e. no `@see` without a `@link`?

JeanMeche replied:
It seems a great addition to me since `@link` also allows external links (we have one external link with `@see`). 👍

devversion replied:
do you think we can do this in this PR?

JeanMeche replied:
Yes, I'll take a look !

JeanMeche replied:
I choose to only throw when backticks are used. We have valid `@see` with direct links to the guides.

devversion reviewed (approved):
This work is greatly appreciated. I could global-approve, but I have one comment

JeanMeche committed:
docs(docs-infra): throw an error when @link is used with a backtick reference

As part of #50097, we'll add this constraint to the @see comments

JeanMeche committed:
docs(localize): Make links out of @see tags

This commit is part of the work for angular#50097 to improve the linking on the online documentation.

JeanMeche committed:
docs(platform-server): Make links out of @see tags

This commit is part of the work for angular#50097 to improve the linking on the online documentation.

devversion reviewed (approved):
LGTM

Reviewed-for: global-dev-infra-approvers

pkozlowski-opensource commented:
This PR was merged into the repository by commit f199305db6340da3ac7520bd29d3208ee9906924.
